### PR TITLE
Fix p18 issue For an item with P18 item, do not add another one

### LINKED ISSUE #3333: For an item that already has a P18, do not add another one
The Nearby activity now also shows Wikidata items that **have** a P18 image. That's great.

But when the user choose to add an image based on this item, the image should not be added as an another P18.
Why? Because a Wikidata item should only have one P18 image. By the way, the selection of the most representative image is sometimes the result of a community conversation, so replacing the existing image is probably not a good idea either.

The item's information can still be used to propose categories/etc, though, so it makes sense to keep the + even for items that already have a P18.

I tag this as a bug because I have to clean up manually behind haha.

### PR DESCRIPTION
**Pull request by neslihanturan**
**Description (required)**

Fixes #3333 For an item with P18 item, do not add another one.
It checks if an item has image or not. If it has an image (green dots on the map) then even if we upload from the map, our upload is a regular commons upload (does not edit wikidata image property). If the wikidata item doesnt have the image, then our upload form map will fill the image property.

**Tests
Tested on a real device, wit android 9. Both cases are tested with upload flows, they worked as expected.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java [hunk 1]
```diff
@@ -68,6 +68,7 @@ public Contribution createFromParcel(Parcel parcel) {
     public String decimalCoords;
     public boolean isMultiple;
     public String wikiDataEntityId;
+    public String p18Value;
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java]: Why is it public when it has a getter and a setter?

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java]: Very true, sorry ı just ctrl+D from previous line, will fix it ASAP:)

File: app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditService.java [hunk 2]
```diff
@@ -64,6 +64,11 @@ public void createClaimWithLogging(String wikidataEntityId, String fileName) {
             return;
         }
 
+        if (!p18Value.trim().isEmpty()) {
+            Timber.d("Skipping creation of claim as p18Value is not null, we won't override existing image");
```
Comment (macgills) [inline app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditService.java]: This log is lying, this isn't "not null" it is "not blank"(to borrow the expression from the kotlin standard library), if this method accepts null values for the p18 then this is a guaranteed NPE

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditService.java]: Yeah, fixed:) it is never null, empty or value

Comment (macgills) [reply app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditService.java]: Excellent, please mark the parameter `@NonNull`

Comment (neslihanturan) [reply app/src/main/java/fr/free/nrw/commons/wikidata/WikidataEditService.java]: Done @macgills thanks a lot:)

**Comment by misaochan** [comment]
Tested on real device running Android 10, works well for me. Thanks @neslihanturan !

**Pull request merged**